src/views/requestAccess.js

```diff
@@ -1,146 +1,123 @@
 			<hr className="line" ></hr>
 			<span className="input-field"> {label}
-				<p className={infoField} contentEditable>
-			        {info}
-			    </p>
+				<input className={`contact-input ${infoField}`} type="text" defaultValue={info} onChange={(e) => changeHandler(e.target.value)}>
+			    </input>
 		    </span>
 	    </div>
 	);
 }
 
 export const RequestAccess = (props) => {
-	// const id = props.match.params.id;
+  // Get context for API auth header
+  const userContext = useContext(UserContext);
+
+  const [roleObject, setRoleObject] = useState({});
+  const [selectionOptions, setSelectionOptions] = useState([]);
+  const [currentSelection, selectionHandler] = useState("");
+  const [fName, setFirstName] = useState("");
+  const [lName, setLastName] = useState("");
+  const [emailAddress, setEmail] = useState("");
+
+  useEffect(() => {
+    console.log(process.env.REACT_APP_API_URL)
+    axios.get(`/api/roles`)
+    .then((response) => {
+      let data = JSON.parse(response.data);
+      // Get key value object of all roles
+      setRoleObject(data);
+      // Get Role names
+      let roleArray = Object.keys(data);
+      // Removes "Pending" role
+      roleArray.shift();
+      // Get array of roles to map to selection options
+      setSelectionOptions(roleArray);
+    })
+    .catch((error) => {
+      alert(error);
+      console.log(error);
+    });
+  }, []);
+
+  const grantAccess = (role, firstName, lastName, email, id) => {
+    // Get role index from roleObject
+    let roleID = roleObject[role];
+    axios.patch(`/api/user/${id}`, {
+      role: roleID,
+      firstName: firstName,
+      lastName: lastName,
+      email: email
+    }, makeAuthHeaders(userContext))
+    .then((response) => {
+      alert("User access granted!")
+    })
+    .catch((error) => {
+      alert(error);
+      console.log(error);
+    });
+  }
 
 	const {
 		firstName,
 		lastName,
-		email
-	} = props.location.state;
+    email,
+    id
+  } = props.location.state;
+  
+  useEffect(() => {
+    setFirstName(firstName);
+    setLastName(lastName);
+    setEmail(email);
+  }, []);
 
 	return (
 		<>
 			<div className="request-page">
 				<div className="page-title"> Request Access </div>
 				<div className="sub-title"> CONTACT </div>
-				<InfoField label={"First Name"} info={firstName} />
-				<InfoField label={"Last Name"} info={lastName} />
+				<InfoField label={"First Name"} changeHandler={setFirstName} info={firstName} />
+				<InfoField label={"Last Name"} changeHandler={setLastName} info={lastName} />
 				{/* <InfoField label={"Phone"} info={data['phone']} /> */}
-				<InfoField label={"Email"} info={email} />
+				<InfoField label={"Email"} changeHandler={setEmail} info={email} />
 				<hr className="line" ></hr>
 				<div className="sub-title sub-title-padding"> ASSIGN ROLE </div>
-				<RoleDropDown />
+				<RoleDropDown selectionOptions={selectionOptions} selectionHandler={selectionHandler}/>
 				<div className="button-padding">
```

Comment:
Let's make sure there's an issue created to add the "Property" section like in the mockup, or we can do that work with this PR! Up to you
![image](https://user-images.githubusercontent.com/8600792/90592319-29293100-e19a-11ea-9779-17ce71e490df.png)